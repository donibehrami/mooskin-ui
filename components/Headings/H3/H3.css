@import url('https://fonts.googleapis.com/css?family=Convergence');
@import '../../_utils/globals/variables.css';
@import '../../_utils/globals/screens.css';

.h3 {
    font-family: 'Convergence', sans-serif;
    /*font-size: 17px;*/
    font-size: 1.063rem;
    font-weight: 400;
    color: var(--heading);
}

@media (--small){
    .h3 {
        font-size: 3.3vw;
    }
}

@media (--xSmall){
    .h3 {
        font-size: 3.3vw;
    }
}