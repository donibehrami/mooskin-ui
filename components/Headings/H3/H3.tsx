import * as React from 'react';

import styles from './H3.css';

import {IHeadingProps} from '../props';

class H3 extends React.Component<IHeadingProps, {}> {

    static defaultProps = {
        className: '',
        style: {}
    };

    render() {

        const {id, className, style, children} = this.props;

        const classes = `h3-component ${styles.h3} ${className}`;

        return (
            <div
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </div>
        );
    }
}

export default H3;
