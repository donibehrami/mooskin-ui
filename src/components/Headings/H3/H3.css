@import url('https://fonts.googleapis.com/css?family=Convergence');

.h3 {
    font-family: 'Convergence', sans-serif;
    font-size: 17px;
    font-weight: 400;
    color: #3a4257;
}