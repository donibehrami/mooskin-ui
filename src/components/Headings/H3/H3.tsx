import * as React from 'react';

import styles from './H3.css';

import {IHeadingProps} from '../../../types/commonTypes';

class H3 extends React.Component<IHeadingProps, void> {

    public static defaultProps = {
        className: '',
        style: {}
    };

    public render() {

        const {id, className, style, children} = this.props;

        const classes = `button-component ${styles.h3} ${className}`;

        return (
            <h3
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </h3>
        );
    }
}

export default H3;
