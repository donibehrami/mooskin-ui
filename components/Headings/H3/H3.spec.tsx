import * as React from 'react';
import H3 from './H3';

import { shallow } from 'enzyme';

describe('H3', () => {

    test('renders correctly', () => {

        const tree = shallow(
            <H3
                className="myClass"
                style={{color: 'blue'}}
                id={'heading3'}
            >
                Mooskin
            </H3>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders properly into dom with text, id and class', () => {

        const component = shallow(<H3 id="Campaigns" className="headers">Mooo!</H3>);

        expect(component.find('p').text()).toBe('Mooo!');
        expect(component.find('p').prop('id')).toEqual('Campaigns');
        expect(component.find('p').hasClass('headers')).toBe(true);
    });

});
