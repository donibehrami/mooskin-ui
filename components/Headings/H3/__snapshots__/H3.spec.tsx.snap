// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 renders correctly 1`] = `
<p
  className="h3-component h3 myClass"
  id="heading3"
  style={
    Object {
      "color": "blue",
    }
  }
>
  Mooskin
</p>
`;
